pkg/agent/metrics/metrics.go

```diff
@@ -46,49 +48,158 @@ var (
 
 // AgentMetrics includes all the metrics of the proxy agent.
 type AgentMetrics struct {
-	latencies *prometheus.HistogramVec
-	failures  *prometheus.CounterVec
+	dialLatencies       *prometheus.HistogramVec
+	serverFailures      *prometheus.CounterVec
+	dialFailures        *prometheus.CounterVec
+	serverConnections   *prometheus.GaugeVec
+	endpointConnections *prometheus.GaugeVec
+	streamEvents        *prometheus.CounterVec
+	streamEventsErrors  *prometheus.CounterVec
 }
 
 // newAgentMetrics create a new AgentMetrics, configured with default metric names.
 func newAgentMetrics() *AgentMetrics {
-	latencies := prometheus.NewHistogramVec(
+	dialLatencies := prometheus.NewHistogramVec(
 		prometheus.HistogramOpts{
-			Namespace: namespace,
-			Subsystem: subsystem,
+			Namespace: Namespace,
+			Subsystem: Subsystem,
 			Name:      "dial_duration_seconds",
 			Help:      "Latency of dial to the remote endpoint in seconds",
 			Buckets:   latencyBuckets,
 		},
 		[]string{},
 	)
-	failures := prometheus.NewCounterVec(
+	serverFailures := prometheus.NewCounterVec(
 		prometheus.CounterOpts{
-			Namespace: namespace,
-			Subsystem: subsystem,
+			Namespace: Namespace,
+			Subsystem: Subsystem,
 			Name:      "server_connection_failure_count",
```

Comment:
This should be `_total`. Is this a new metric?

Reply:
No, server_connection_failure_count is not new.